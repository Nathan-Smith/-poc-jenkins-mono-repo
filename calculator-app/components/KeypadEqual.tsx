import { KeypadButton } from './KeypadButton'

export function KeypadEqual({
  onInput,
}: {
  onInput?: () => void
}): JSX.Element {
  return (
    <KeypadButton
      testid="keypad-equal"
      gridArea="keypad-equal"
      onClick={() => {
        onInput?.()
      }}
    >
      =
    </KeypadButton>
  )
}
