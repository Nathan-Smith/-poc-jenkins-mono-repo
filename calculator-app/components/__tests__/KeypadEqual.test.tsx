import React from 'react'
import renderer, { act, ReactTestRenderer } from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react'

import { KeypadEqual } from '../KeypadEqual'

describe('KeypadEqual', () => {
  test('renders', () => {
    let component: ReactTestRenderer | undefined
    act(() => {
      component = renderer.create(<KeypadEqual />)
    })

    expect(component?.toJSON()).toMatchSnapshot()
  })

  test('calls onInput on click', () => {
    const onInput = jest.fn()
    const { getByTestId } = render(<KeypadEqual onInput={onInput} />)

    fireEvent.click(getByTestId(/keypad-equal/i))

    expect(onInput).toBeCalled()
  })
})
