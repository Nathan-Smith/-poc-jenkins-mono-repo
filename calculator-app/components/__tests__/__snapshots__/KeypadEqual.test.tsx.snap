// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeypadEqual renders 1`] = `
<button
  className="mdc-button mdc-button--outlined keypadButton"
  data-testid="keypad-equal"
  onClick={[Function]}
  style={
    Object {
      "gridArea": "keypad-equal",
    }
  }
>
  <span
    className="mdc-button__ripple"
  />
  <span
    className="mdc-button__label"
  >
    =
  </span>
</button>
`;
